pkg/app/appserver/proc_manager.go

```diff
@@ -101,7 +97,7 @@ func (m *procManager) Stop(name string) error {
 
 // Wait waits for the application to exit.
 func (m *procManager) Wait(name string) error {
-	p, err := m.pop(name)
+	p, err := m.get(name)
```

Comment:
Why can't we just pop and then work with the `proc`? we're popping in all of the workflow branches anyway

Reply:
@Darkren It might look like we can, however, we need to have the app in the process list while blocking `p.Wait()` call to maintain the list of currently running apps. It might look not clear, so I think it's worth to add a comment. Thanks!